content/_code-samples/escrow/js/cancel-escrow.js

```diff
@@ -1,40 +1,94 @@
-'use strict'
-const RippleAPI = require('ripple-lib').RippleAPI
+if (typeof module !== "undefined") {
+    // Use var here because const/let are block-scoped to the if statement.
+    var xrpl = require('xrpl')
+  }
+  // Construct an EscrowCancel transaction to cancel an Escrow which has passed its CancelAfter due date
+  // https://xrpl.org/escrow.html#escrow
+  // https://xrpl.org/escrowcancel.html#escrowcancel
 
-const myAddr = 'rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn'
-const mySecret = 's████████████████████████████'
+  async function main() {
+    // To cancel an Escrow, you'd need to have an account which has sent an Escrow with a 'CancelAfter' field pass its time
+    // You could use 'create-escrow.js' on the same folder to import an account
 
-const myEscrowCancellation = {
-  owner: myAddr,
-  escrowSequence: 366
-}
-const myInstructions = {
-  maxLedgerVersionOffset: 5
-}
+    seed = ""
+
+    // Connect to a testnet node
+    console.log("Connecting to Testnet...")
+    const client = new xrpl.Client('wss://s.altnet.rippletest.net:51233')
+    await client.connect()
+
+    wallet = xrpl.Wallet.fromSeed(seed)
+
+    console.log(" Sending Account:", wallet.address)
+    console.log("            Seed:", wallet.seed)
+
+    // Query the account's objects (Type: Escrow)
+    const response = await client.request({
+        "command": "account_objects",
+        "account": wallet.address,
+        "ledger_index": "validated",
+        "type": "escrow"
+    })
+
+    // Only append Escrows that were sent by this Account, incoming Escrows are ignored since we have no authority to cancel them
+    var outgoing = []
+    for (var i = 0; i < response.result.account_objects.length; i++) {
+        if (response.result.account_objects[i].Account == wallet.address) {
+            outgoing.push(response.result.account_objects[i])
+        }
+    }
+
+    // Get current datetime
+    const date = new Date();
+
+    // Only append outgoing Escrows that have passed their CancelAfter time
+    var outgoing_cancelable = []
+    for (var i = 0; i < outgoing.length; i++) {
+        const CancelAfter_date = new Date(xrpl.rippleTimeToISOTime(outgoing[i].CancelAfter));
+        if (CancelAfter_date.getTime() <= date.getTime()) {
+            outgoing_cancelable.push(outgoing[i])
+        }
+    }
```

Comment:
Same as with finish-escrow, we should have clearer next steps if:
1. There are no escrows at all (Wrong account)
2. There are escrows, but they're not cancelable yet (Let the user know how long until one of them is ready)